Remove dead code, cmake cleanups

jagerman opened the pull request:
- Remove shell hooks.  Unused, probably bitrotten, and not wanted anymore.
- Remove netns code.  Unused, bitrotten, and we can pull it out of git history if we want to resurrect it.
- Remove '=1' from compiler definitions (because we only check for definedness, not value, and someone might get the impression that they can set it to 0 to disable it, which they can't).

jagerman committed:
Remove dead code: netns, shell hooks

These haven't been activated in a long time and aren't worth
resuscitating.

jagerman committed:
Cmake cleanup: remove unneeded =1 from definitions

We only check for definedness, not truth, in the code so make the cmake
definitions agree with that.

This also avoids warnings when building on macos (because swift only
allowed defined/undefined but not values)

jagerman committed:
Remove shadow testing framework

Bitrotten and apparently doesn't work with libuv event loop.

majestrate commented on llarp/service/endpoint.hpp:
```diff
@@ -7,6 +7,7 @@
 #include <llarp/path/path.hpp>
 #include <llarp/path/pathbuilder.hpp>
 #include "address.hpp"
+#include "endpoint_state.hpp"
```
i wonder if we could not require this header and forward declare as much as we can.

jagerman replied:
This was required because I removed the destructor, but that meant the implicit one needs to be in the header.  I've made the destructor explicit again and put a defaulted one in the .cpp, which solves this.

jagerman committed:
Move destructor back to .cpp file

Having it there (even defaulted, like this) means endpoint.hpp doesn't
have to include endpoint_state.hpp (which it otherwise would need,
because of the std::unique_ptr<EndpointState> default deleter
requirements).
